p11-kit/test-list-mechanisms.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+test "${abs_top_builddir+set}" = set || {
+	echo "set abs_top_builddir" 1>&2
+	exit 1
+}
+
+. "$abs_top_builddir/common/test-init.sh"
+
+setup() {
+	testdir=$PWD/test-mechanisms-$$
+	test -d "$testdir" || mkdir "$testdir"
+	cd "$testdir"
+}
+
+teardown() {
+	rm -rf "$testdir"
+}
+
+test_list_mechanisms() {
+	cat > list.exp <<EOF
+token: TEST LABEL
+0x80000001 (unknown): encrypt decrypt key-size=512-4096
+0x80000002 (unknown): sign verify key-size=2048-2048
+token: TEST LABEL
+0x80000001 (unknown): encrypt decrypt key-size=512-4096
+0x80000002 (unknown): sign verify key-size=2048-2048
+token: TEST LABEL
+0x80000001 (unknown): encrypt decrypt key-size=512-4096
+0x80000002 (unknown): sign verify key-size=2048-2048
+token: TEST LABEL
+0x80000001 (unknown): encrypt decrypt key-size=512-4096
+0x80000002 (unknown): sign verify key-size=2048-2048
+EOF
+
+	if ! "$abs_top_builddir"/p11-kit/p11-kit-testable list-mechanisms -q pkcs11: > list.out; then
```

Comment:
This URL matches all tokens? I would rather limit it to the first match, as `p11tool --list-mechanism` does.

Reply:
> This URL matches all tokens?

Yes.

> I would rather limit it to the first match, as p11tool --list-mechanism does.

You mean, limit `list-mechanisms` subcommand to inspecting the first matching token? I'd disagree on that. First, I think it might introduce confusion in a scenario where the user accidentally matches several tokens. Second, AFAICS this behaviour won't be in line with the existing `list-*` subcommands that iterate over all matches.

Reply:
> AFAICS this behaviour won't be in line with the existing list-* subcommands that iterate over all matches

Not really:
- `p11-kit list-modules`: doesn't take URL
- `p11-kit list-profiles`: takes a token URL, uses first match only, because profiles are capabilities of a particular token
- `p11-kit list-objects`: takes an *object* URL, returns all matches

Reply:
Changed to stop after first match. Testing 3 scenarios now: "pkcs11:", specific existing and specific non-existing token. Maybe when we have list-tokens we should also test the iteration you're suggesting.